tests/integration/s2n_handshake_test_s_client.py

```diff
@@ -808,23 +864,38 @@ def main():
     if use_corked_io == True:
         print("Corked IO is on")
 
+    results = {'tests_ran': 0}
+
     try:
-        resume_test(host, port, test_ciphers, fips_mode, no_ticket=True)
-        resume_test(host, port, test_ciphers, fips_mode)
-        handshake_test(host, port, test_ciphers, fips_mode)
-        client_auth_test(host, port, test_ciphers, fips_mode)
-        sigalg_test(host, port, fips_mode)
-        sigalg_test(host, port, fips_mode, use_client_auth=True, no_ticket=True)
-        elliptic_curve_test(host, port, libcrypto_version, fips_mode)
-        elliptic_curve_fallback_test(host, port, fips_mode)
-        handshake_fragmentation_test(host, port, fips_mode)
-        ocsp_stapling_test(host, port, fips_mode)
-        cert_type_cipher_match_test(host, port, libcrypto_version)
-        multiple_cert_type_test(host, port, libcrypto_version)
-        multiple_cert_domain_name_test(host, port)
+        for tls13_flag in [False, True]:
+            # options are additional arguments that eventually gets passed to try_handshake()
+            options = dict(
+                results=results,
+                tls13_flag=tls13_flag,
+                debug_cmds=args.debug,
+                simulate_tests=args.dryrun,
+            )
+            if tls13_flag:
```

Comment:
So this is just running the test suite once with and without TLS1.3? It might actually be nice to have it more fine grained, as our Java integration test is => Run the test with specific TLS versions and cipher combinations.

But for the moment this seems fine too.

Reply:
TLS 1.3 tests is currently in s2n_tls13_handshake_tests.py. Here is the TLS 1.2 suites and twice, once the existing approach, and second with --tls13 flag that potentially exercise TLS 1.2 negotiations with potential TLS 1.3 code paths.